build_psa_compliance.py

```diff
@@ -65,15 +65,15 @@ def _run_ff_prerequisites():
     TF-M can update TF-M database with PSA test manifests.
     """
     manifest_dir = join(TF_M_BUILD_DIR, 'psa-arch-tests', 'api-tests')
-    cmd = ['python', 'tools/scripts/manifest_update.py']
+    cmd = ['python3', 'tools/scripts/manifest_update.py']
```

Comment:
Should use `os.path` or `pathlib` here so the code is cross-platform compatible.

Reply:
We are running it as a command from `manifest_dir` from the above case.

Reply:
We should still avoid hard-coding the path separators. `os.path` will take care of that for us and is guaranteed to work cross platform.

Reply:
Actually, `python3` won't work on Windows either, if we care about Windows.

Reply:
Why python3 won't work?
```
C:\WINDOWS\system32>python3
Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 22:22:05) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>
```

where to use os.path? Those scripts are supposed to be executed from a specific path.

Reply:
`python3` isn't traditionally available on Windows. Plausible that's changed. I'll have to double check as I don't use Windows often. 

`os.path` will just format your path string properly. I don't see how executing it from a specific path is relevant to the formatting.

`cmd = ['python3', os.path.join('tools', 'scripts', 'manifest_update.py')]`

Reply:
oh, got it.

Reply:
I think `python3` is only available with newer versions of python installed from Microsoft Store. Seems fragile to assume it's always there.

If we could avoid executing python as a subprocess from within a python script we really should.

Reply:
The script `build_psa_compliance.py` is supposed to be executed in the python3 environment. As there were few issues with encoding observed during testing.
I'm unsure if `import` is possible.

Reply:
OK. This is probably because the existing mbed tools execution model is messed up. Hopefully we can fix this kind of thing with the new tools.